File: agora-ts/packages/core/src/enums.ts
export const TaskState = {
  DRAFT: 'draft',
  CREATED: 'created',
  ACTIVE: 'active',
  BLOCKED: 'blocked',
  PAUSED: 'paused',
  DONE: 'done',
  CANCELLED: 'cancelled',
  ORPHANED: 'orphaned',
} as const;

export type TaskState = (typeof TaskState)[keyof typeof TaskState];

export const GateType = {
  ARCHON_REVIEW: 'archon_review',
  COMMAND: 'command',
  ALL_SUBTASKS_DONE: 'all_subtasks_done',
  APPROVAL: 'approval',
  AUTO_TIMEOUT: 'auto_timeout',
  QUORUM: 'quorum',
} as const;

export type GateType = (typeof GateType)[keyof typeof GateType];

export const SubtaskState = {
  PENDING: 'pending',
  IN_PROGRESS: 'in_progress',
  WAITING_INPUT: 'waiting_input',
  DONE: 'done',
  FAILED: 'failed',
  CANCELLED: 'cancelled',
  ARCHIVED: 'archived',
} as const;

export type SubtaskState = (typeof SubtaskState)[keyof typeof SubtaskState];

export const CollaborationMode = {
  DISCUSS: 'discuss',
  INDEPENDENT_EXECUTE: 'execute',
} as const;

export type CollaborationMode = (typeof CollaborationMode)[keyof typeof CollaborationMode];

export const AgentRole = {
  ARCHITECT: 'architect',
  DEVELOPER: 'developer',
  REVIEWER: 'reviewer',
  WRITER: 'writer',
  RESEARCHER: 'researcher',
  ANALYST: 'analyst',
  EXECUTOR: 'executor',
  CRAFTSMAN: 'craftsman',
} as const;

export type AgentRole = (typeof AgentRole)[keyof typeof AgentRole];

export const DispatchStatus = {
  QUEUED: 'queued',
  SUCCESS: 'success',
  FAILED: 'failed',
  TIMEOUT: 'timeout',
  GATEWAY_OFFLINE: 'gateway_offline',
  ERROR: 'error',
} as const;

export type DispatchStatus = (typeof DispatchStatus)[keyof typeof DispatchStatus];

export const EscalationLevel = {
  NONE: 0,
  RETRY: 1,
  NOTIFY: 2,
  ROLLBACK: 3,
  HUMAN: 4,
} as const;

export type EscalationLevel = (typeof EscalationLevel)[keyof typeof EscalationLevel];

export const ActivityKind = {
  FLOW: 'flow',
  PROGRESS: 'progress',
  TODOS: 'todos',
  ASSISTANT: 'assistant',
  TOOL_RESULT: 'tool_result',
  USER: 'user',
  SYSTEM: 'system',
  ARCHON: 'archon',
} as const;

export type ActivityKind = (typeof ActivityKind)[keyof typeof ActivityKind];

export const GovernancePreset = {
  LEAN: 'lean',
  STANDARD: 'standard',
  STRICT: 'strict',
  CUSTOM: 'custom',
} as const;

export type GovernancePreset = (typeof GovernancePreset)[keyof typeof GovernancePreset];

export const CraftsmanType = {
  CLAUDE_CODE: 'claude_code',
  CODEX: 'codex',
  GEMINI_CLI: 'gemini_cli',
  CUSTOM: 'custom',
} as const;

export type CraftsmanType = (typeof CraftsmanType)[keyof typeof CraftsmanType];

export const TaskType = {
  CODING: 'coding',
  CODING_HEAVY: 'coding_heavy',
  RESEARCH: 'research',
  DOCUMENT: 'document',
  QUICK: 'quick',
  BRAINSTORM: 'brainstorm',
  CUSTOM: 'custom',
} as const;

export type TaskType = (typeof TaskType)[keyof typeof TaskType];

export const TaskPriority = {
  HIGH: 'high',
  NORMAL: 'normal',
  LOW: 'low',
} as const;

export type TaskPriority = (typeof TaskPriority)[keyof typeof TaskPriority];

File: archive/agora-python-legacy/core/enums.py
"""Agora canonical enums — single source of truth.

All enum values MUST match docs/01-PLANS/ENUMS.md exactly.
"""
from enum import Enum


class TaskState(str, Enum):
    """任务顶层状态"""
    DRAFT = "draft"
    CREATED = "created"
    ACTIVE = "active"
    BLOCKED = "blocked"
    PAUSED = "paused"
    DONE = "done"
    CANCELLED = "cancelled"
    ORPHANED = "orphaned"


class SubtaskState(str, Enum):
    """子任务状态"""
    NOT_STARTED = "not_started"
    DISPATCHED = "dispatched"
    IN_PROGRESS = "in_progress"
    DONE = "done"
    FAILED = "failed"
    RETRYING = "retrying"
    ESCALATED = "escalated"


class CollaborationMode(str, Enum):
    """协作模式"""
    DISCUSS = "discuss"
    INDEPENDENT_EXECUTE = "execute"


class GateType(str, Enum):
    """Gate 类型"""
    ARCHON_REVIEW = "archon_review"
    COMMAND = "command"
    ALL_SUBTASKS_DONE = "all_subtasks_done"
    APPROVAL = "approval"
    AUTO_TIMEOUT = "auto_timeout"
    QUORUM = "quorum"


class AgentRole(str, Enum):
    """Agent 角色"""
    ARCHITECT = "architect"
    DEVELOPER = "developer"
    REVIEWER = "reviewer"
    WRITER = "writer"
    RESEARCHER = "researcher"
    ANALYST = "analyst"
    EXECUTOR = "executor"
    CRAFTSMAN = "craftsman"


class DispatchStatus(str, Enum):
    """调度状态"""
    QUEUED = "queued"
    SUCCESS = "success"
    FAILED = "failed"
    TIMEOUT = "timeout"
    GATEWAY_OFFLINE = "gateway_offline"
    ERROR = "error"


class EscalationLevel(int, Enum):
    """升级级别"""
    NONE = 0
    RETRY = 1
    NOTIFY = 2
    ROLLBACK = 3
    HUMAN = 4


class ActivityKind(str, Enum):
    """活动类型"""
    FLOW = "flow"
    PROGRESS = "progress"
    TODOS = "todos"
    ASSISTANT = "assistant"
    TOOL_RESULT = "tool_result"
    USER = "user"
    SYSTEM = "system"
    ARCHON = "archon"


class GovernancePreset(str, Enum):
    """治理预设"""
    LEAN = "lean"
    STANDARD = "standard"
    STRICT = "strict"
    CUSTOM = "custom"


class CraftsmanType(str, Enum):
    """工匠类型"""
    CLAUDE_CODE = "claude_code"
    CODEX = "codex"
    GEMINI_CLI = "gemini_cli"
    CUSTOM = "custom"


class TaskType(str, Enum):
    """任务类型"""
    CODING = "coding"
    CODING_HEAVY = "coding_heavy"
    RESEARCH = "research"
    DOCUMENT = "document"
    QUICK = "quick"
    BRAINSTORM = "brainstorm"
    CUSTOM = "custom"


class TaskPriority(str, Enum):
    """任务优先级"""
    HIGH = "high"
    NORMAL = "normal"
    LOW = "low"
